File: src/main/java/com/ivanqueiroz/curriculo/model/Curriculo.java
package com.ivanqueiroz.curriculo.model;

import java.io.Serializable;
import java.util.List;
import javax.persistence.CollectionTable;
import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import lombok.Data;

/**
 *
 * @author Ivan Queiroz <ivanqueiroz@gmail.com>
 */
@Data
@Entity
public class Curriculo implements Serializable {

    private static final long serialVersionUID = -2900842519899417795L;

    @Id
    @GeneratedValue
    private Long id;
    private String nome;
    private String profissao;
    private String resumo;

    @OneToMany
    private List<Instituicao> academico;

    @OneToMany
    private List<Certificacao> certificacoes;

    @OneToMany
    private List<Cargo> experiencia;

    @OneToMany
    private List<Contato> contatos;

    @OneToMany
    private List<Conhecimento> conhecimentos;

    @OneToMany
    private List<Lingua> linguas;

    @ElementCollection
    @CollectionTable(name = "Passatempos", joinColumns = @JoinColumn(name = "curriculo_id"))
    private List<String> passatempos;

    public String getNomeCurto() {
        if (nome != null && !nome.isEmpty()) {
            String[] nomes = nome.split(" ");
            if (nomes.length > 1) {
                return nomes[0] + " " + nomes[nomes.length - 1];
            }
        }
        return nome;
    }

}

File: src/main/kotlin/dev/ivanqueiroz/curriculo/dominio/curriculo/Curriculo.kt
package dev.ivanqueiroz.curriculo.dominio.curriculo

import dev.ivanqueiroz.curriculo.dominio.conhecimento.Conhecimento
import dev.ivanqueiroz.curriculo.dominio.contato.Contato
import dev.ivanqueiroz.curriculo.dominio.historico.Historico
import javax.persistence.*

@Entity
data class Curriculo(
        @Id
        var id: Long = 0L,

        @Column(length = 100, nullable = false, name = "PRIMEIRO_NOME")
        var primeiroNome: String = "",

        @Column(length = 100, nullable = false, name = "ULTIMO_NOME")
        var ultimoNome: String = ""
) {
    @Column(length = 240, nullable = false)
    var resumo: String = "";

    @Column(length = 240, nullable = false)
    var profissao: String = ""

    @OneToMany(cascade = [CascadeType.ALL])
    var historicos: List<Historico>? = null

    @OneToMany(cascade = [CascadeType.ALL])
    var contatos: List<Contato>? = null

    @OneToMany(cascade = [CascadeType.ALL])
    var conhecimentos: List<Conhecimento>? = null
}